services/gitdiff/gitdiff.go

```diff
@@ -854,7 +856,13 @@ func GetDiffRangeWithWhitespaceBehavior(repoPath, beforeCommitID, afterCommitID
 	defer cancel()
 	var cmd *exec.Cmd
 	if (len(beforeCommitID) == 0 || beforeCommitID == git.EmptySHA) && commit.ParentCount() == 0 {
-		cmd = exec.CommandContext(ctx, git.GitExecutable, "show", afterCommitID)
+		diffArgs := []string{"diff", "--src-prefix=\\a/", "--dst-prefix=\\b/", "-M"}
+		if len(whitespaceBehavior) != 0 {
+			diffArgs = append(diffArgs, whitespaceBehavior)
+		}
+		diffArgs = append(diffArgs, "4b825dc642cb6eb9a060e54bf8d69288fbee4904")
```

Comment:
This probably should be beforeCommitID?

Reply:
Nope. That is the beforecommitid in this case

The commit has no parent so we use the magic empty tree ref.

Believe it or not git this is what git uses internally to generate the "diff" on the git show in cases where the commit has no parents

Reply:
We can't set this as the beforecommitid as it breaks other things lower down in the function.